### PR TITLE
Make rtest fail less than 1 in 20

### PR DESCRIPTION
**Pull request by originalsouth**
### Changes
Patches for Celery and Robot files.

### Issue link
Address issue #2336 

Closes ...

### Demo
N/A (see pipeline, hopefully)

---

### Code Checklist
- [x] All the commits in this PR are properly PGP-signed and verified.
- [x] This PR only contains functionality relevant to the issue; tickets have been created for newly discovered issues.
- [x] I have written unit tests for the changes or fixes I made.
- [x] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

### Communication
- [x] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [x] I have made corresponding changes to the documentation, if necessary.
- [x] I have included comments in the code to elaborate on what is not self-evident from the code itself, including references to issues and discussions online, or implicit behavior of an interface.

---
## Checklist for code reviewers:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---
## Checklist for QA:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

**Comment by originalsouth** [comment]
Thanks for the feedback. I agree. `Wait Until Keyword Succeed` would be a better - indeed. I will experiment with it later  today/this week. 

The underlying problem here, also effecting #2083, is however that (re)calculating the bits takes a relatively rather large and inconsistent number of seconds while also being asynchronous. Incidentally, this is any issue which I am currently investigating.

**Comment by dekkers** [comment]
I first thought there was a race condition in the tests, but when working to change the tests I saw that it actually does wait for all celery tasks to execute. How did you conclude that the underlying problem is the number of seconds recalculation takes?

Did you see that there were still tasks being executed or waiting to be executed after "Await Sync" (which calls "Await Tasks Done")?  If so we should fix "Await Tasks Done" instead of adding sleep statements.

The problem might also not be in the tests.. One strange thing I saw is that even after I added sleep statements of 10 seconds the number of tasks that were executed as reported by robotframework/celerymonitor logging wasn't stable in the test I was looking at even after I disabled celery beat. For example one time it was 92, then 95, then 93, etc. There might be race conditions in how the bits and scan profile propagation are executed in Octopoes itself instead of in the tests.

**Comment by originalsouth** [comment]
> How did you conclude that the underlying problem is the number of seconds recalculation takes?

Similarly to #2083 the error occurs after a recalculation, adding a sleep "resolves" the problem indicating a timing issue in recalculation. I think on of the problems is that bits can trigger new bits recursively and we do not block till this process is finished. With the event manager mock in #2310 we dig deeper into the this issue (although high resolution time stamps might still be required).

File: octopoes/tests/robot/05_bits.robot [hunk 1]
```diff
@@ -12,8 +14,8 @@ Bit With Scan Level 1
     Object List Should Not Contain    Finding|Hostname|internet|example.com|KAT-NXDOMAIN
     Declare Scan Profile    Hostname|internet|example.com    1
     Await Sync
-    Object List Should Contain    KATFindingType|KAT-NXDOMAIN
-    Object List Should Contain    Finding|Hostname|internet|example.com|KAT-NXDOMAIN
+    Wait Until Keyword Succeeds    60s    1s    Object List Should Contain    KATFindingType|KAT-NXDOMAIN
+    Wait Until Keyword Succeeds    60s    1s    Object List Should Contain    Finding|Hostname|internet|example.com|KAT-NXDOMAIN
```
Comment (dekkers) [inline octopoes/tests/robot/05_bits.robot:18]: Does this still fail the test if it isn't there after 60 seconds? The robot framework documentation isn't very clear but I think they mean to say that it doesn't.

File: octopoes/tests/robot/CeleryMonitor.py [hunk 2]
```diff
@@ -11,13 +11,17 @@
...
+        "worker_concurrency": 1,
+        "worker_prefetch_multiplier": 0,
         "result_backend": f"rpc://{queue_uri}",
         "task_serializer": "json",
         "result_serializer": "json",
         "event_serializer": "json",
         "accept_content": ["application/json", "application/x-python-serialize"],
         "result_accept_content": ["application/json", "application/x-python-serialize"],
         "task_queues": ("octopoes",),
+        "task_alway_eager": True,
```
Comment (dekkers) [inline octopoes/tests/robot/CeleryMonitor.py:24]: This is the config for the celerymonitor / the part running inside the robot tests, so won't do anything as far as I understand it. The tasks are created and executed inside octopoes api and api worker and not the robot tests.

I also think it is better if we don't configure celery in the integration tests differently from how it is normally run unless it is really necessary.

File: octopoes/tests/robot/05_bits.robot (hunk 1 above)
Comment (originalsouth) [reply octopoes/tests/robot/05_bits.robot:18]: Yes, I believe that is the gist of it; see [here](http://robotframework.org/robotframework/2.6.1/libraries/BuiltIn.html).

File: octopoes/tests/robot/CeleryMonitor.py (hunk 2 above)
Comment (originalsouth) [reply octopoes/tests/robot/CeleryMonitor.py]: Addressed in 5dfb5afad4312791bc4b3dc6c6409884247f1425.

**Review by dekkers — APPROVED** [review]

**Pull request merged**